packages/react-components/react-rating-preview/src/components/RatingItem/useRatingItem.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { getPartitionedNativeProps, slot } from '@fluentui/react-utilities';
+import { useFocusWithin } from '@fluentui/react-tabster';
+import type { RatingItemProps, RatingItemState } from './RatingItem.types';
+import { useRatingContextValue_unstable } from '../../contexts/RatingContext';
+import { StarRegular, StarFilled, StarHalfRegular } from '@fluentui/react-icons';
+
+/**
+ * Create the state required to render RatingItem.
+ *
+ * The returned state can be modified with hooks such as useRatingItemStyles_unstable,
+ * before being passed to renderRatingItem_unstable.
+ *
+ * @param props - props from this instance of RatingItem
+ * @param ref - reference to root HTMLElement of RatingItem
+ */
+export const useRatingItem_unstable = (props: RatingItemProps, ref: React.Ref<HTMLInputElement>): RatingItemState => {
+  const context = useRatingContextValue_unstable();
+  const { value = 0 } = props;
+
+  const ratingValue = context.value || 0;
+
+  const displayedRatingValue = context.hoveredValue !== undefined ? context.hoveredValue : ratingValue;
+
+  const nativeProps = getPartitionedNativeProps({
+    props,
+    primarySlotTagName: 'input',
+    excludedPropNames: ['defaultChecked', 'onChange'],
+  });
+
+  const root = slot.always(props.root, {
+    defaultProps: { ref: useFocusWithin<HTMLSpanElement>(), ...nativeProps.root },
+    elementType: 'span',
+  });
+
+  let icon;
+  if (displayedRatingValue && displayedRatingValue >= value) {
```

Comment:
You don't need the test for `displayedRatingValue`. You are guaranteeing it is defined above:

```suggestion
  if (displayedRatingValue >= value) {
```

same for the `else if` below.